reflex-dom-core/src/Reflex/Dom/Builder/Immediate.hs

```diff
@@ -803,9 +808,13 @@ hydrateElement elementTag cfg child = do
               True -> go (Just node) -- this element should be skipped by hydration
               False -> do
                 t <- Element.getTagName e
-                -- TODO: check attributes?
                 if T.toCaseFold elementTag == T.toCaseFold t
-                  then pure e
+                  then do
+                    let shouldPatchAttrs = True -- TODO do something with this flag if hydration is much slower
+                    when shouldPatchAttrs $ do
+                      p <- sample $ currentIncremental incrementalAttrs
+                      liftJSM $ traverse_ (applyAttrPatchDOM @ModifyAttrs PatchSettings_CheckBeforePatching e) $ toPatch p
+                    pure e
```

Comment:
Per our previous discussion, hydration should *never* patch attributes - it should know they're correct from the outset.